minEnclosingTriangle Error using python3.8.3 on Windows

##### System information (version)
- OpenCV => 4.2
- Operating System / Platform => Windows 7 64 Bit
- Python 3.8.3
##### Detailed description
I'm trying to calculate the minEnclosingTriangle, but i'm getting error:

> cv2.error: OpenCV(4.2.0) C:\projects\opencv-python\opencv\modules\core\src\copy.cpp:254: error: (-215:Assertion failed) channels() == CV_MAT_CN(dtype) in function 'cv::Mat::copyTo'

here is my code:
```
import cv2 as cv
import numpy as np

arr = np.array([[0.0, 0.0],[0.0, 1.0],[1.0, 1.0]], dtype=np.float32)
print(arr)
c,r = cv.minEnclosingCircle(arr)
print(c,r)
t,t1 = cv.minEnclosingTriangle(arr)
print(t,t1)
```
and the output:

> λ python triangle.py
> [[0. 0.]
>  [0. 1.]
>  [1. 1.]]
> (0.5, 0.5) 0.7072067856788635
> Traceback (most recent call last):
>   File "triangle.py", line 8, in <module>
>     t,t1 = cv.minEnclosingTriangle(arr)
> cv2.error: OpenCV(4.2.0) C:\projects\opencv-python\opencv\modules\core\src\copy.cpp:254: error: (-215:Assertion failed) channels() == CV_MAT_CN(dtype) in function 'cv::Mat::copyTo'

The circle was calculated, so assume the InputArray points should be OK.
What I'm doing wrong?

--- a/modules/imgproc/src/min_enclosing_triangle.cpp
+++ b/modules/imgproc/src/min_enclosing_triangle.cpp
@@ -318,7 +318,7 @@ namespace minEnclosingTriangle {
 static void findMinEnclosingTriangle(cv::InputArray points,
                                      CV_OUT cv::OutputArray triangle, CV_OUT double &area) {
     std::vector<cv::Point2f> resultingTriangle, polygon;
-    CV_Assert(points.getMat().checkVector(2) > 0);
+    CV_Assert(!points.empty());
     convexHull(points, polygon, true, true);
     findMinEnclosingTriangle(polygon, resultingTriangle, area);
     cv::Mat(resultingTriangle).copyTo(triangle);

--- a/modules/imgproc/src/min_enclosing_triangle.cpp
+++ b/modules/imgproc/src/min_enclosing_triangle.cpp
@@ -129,7 +129,6 @@ static bool areOnTheSameSideOfLine(const cv::Point2f &p1, const cv::Point2f &p2,
 
 static double areaOfTriangle(const cv::Point2f &a, const cv::Point2f &b, const cv::Point2f &c);
 
-static void createConvexHull(cv::InputArray points, std::vector<cv::Point2f> &polygon);
 
 static double distanceBtwPoints(const cv::Point2f &a, const cv::Point2f &b);
 
@@ -319,27 +318,10 @@ namespace minEnclosingTriangle {
 static void findMinEnclosingTriangle(cv::InputArray points,
                                      CV_OUT cv::OutputArray triangle, CV_OUT double &area) {
     std::vector<cv::Point2f> resultingTriangle, polygon;
-
-    createConvexHull(points, polygon);
+    CV_Assert(points.getMat().checkVector(2) > 0);
+    convexHull(points, polygon, true, true);
     findMinEnclosingTriangle(polygon, resultingTriangle, area);
     cv::Mat(resultingTriangle).copyTo(triangle);
-}
-
-//! Create the convex hull of the given set of points
-/*!
-* @param points     The provided set of points
-* @param polygon    The polygon representing the convex hull of the points
-*/
-static void createConvexHull(cv::InputArray points, std::vector<cv::Point2f> &polygon) {
-    cv::Mat pointsMat = points.getMat();
-    std::vector<cv::Point2f> pointsVector;
-
-    CV_Assert((pointsMat.checkVector(2) > 0) &&
-              ((pointsMat.depth() == CV_32F) || (pointsMat.depth() == CV_32S)));
-
-    pointsMat.convertTo(pointsVector, CV_32F);
-
-    convexHull(pointsVector, polygon, true, true);
 }
 
 //! Find the minimum enclosing triangle and its area

